Makefile

```diff
@@ -1,30 +1,40 @@
-.PHONY: docs test clean pypi
+.PHONY: docs test clean pypi env
 
-test:
-	python -m pytest src --doctest-modules -vvv
-	PYTHONPATH="./src/" python -m pytest --cov=./src --cov-report html:htmlcov ./tests -vvv
-	python -m pycodestyle tests
-	python -m pycodestyle src
+PYTHON:=$(shell test -e env/bin/activate && echo "env/bin/python" || echo "python3")
 
-testcoverage:
-	PYTHONPATH="./src/" python -m pytest --cov=./src tests/
+env:
+	echo "Using $(shell exec $(PYTHON) --version): $(PYTHON)"
+	$(PYTHON) -m ensurepip
 
-docs:
-	cd docs/ && make clean html
+test: env lint
+	$(PYTHON) -m pytest src --doctest-modules -vvv
+	PYTHONPATH="./src/" $(PYTHON) -m pytest --cov=./src --cov-report html:htmlcov ./tests -vvv
 
-setupdocs: setuptools
-	python -m pip install -r docs/requirements.txt
+lint:
+	$(PYTHON) -m pycodestyle tests
+	$(PYTHON) -m pycodestyle src
 
-dev: setuptools
-	python -m pip install '.[test]'
+testcoverage: env
+	PYTHONPATH="./src/" $(PYTHON) -m pytest --cov=./src tests/
+
+docs: setupdocs clean
+	cd docs/ && make html
+
+setupdocs: env
+	$(PYTHON) -m pip install -r docs/requirements.txt
+	cd docs/ && yarn install
```

Comment:
Mostly yarn is the preferred package manager for many nodejs packages. The reason I used it here is because it is what mermaid-js itself documents in its install instructions. Since it is only for documentation generation I do not see the extra dependency on yarn as a big issue. 
As far as benefits, there are claims of improved speed, security and stability that I haven't gone deeper into. But truth be told, since it is not part of the main benchmarkstt code, I have no preference either way. The claims of yarn dependency on the mermaid-js installation instructions makes me more inclined to keep using yarn though.

Ref: https://mermaid-js.github.io/mermaid/#/?id=installation

Reply:
(removed the `package-lock.json` as it was a left-over of installing `mermaid-cli` during my first attempt to use mermaid without using the sphinxcontrib-mermaid extension)

Reply:
(removed `yarn install` from this Makefile as it is already being run in the docs/Makefile)

Reply:
I've now factored out yarn completely and used `pynpm` to do the `mermaid` installation.